autotest/conftest.py

```diff
@@ -152,3 +152,17 @@ def pytest_configure(config):
             f"Attempting to run tests for GDAL {test_version} but library version is "
             f"{lib_version}. Do you need to run setdevenv.sh ?"
         )
+
+    import gdaltest
+
+    if not gdaltest.download_test_data():
```

Comment:
or perhaps change the test to:
```suggestion
    if not os.path.exists("gdrivers/tmp/cache") or not os.path.exists("ogr/tmp/cache"):
```

Reply:
Those paths could exist from a past test run, but that doesn't mean that test datasets are available for the current run. Is the concern that we would emit a message about `GDAL_DOWNLOAD_TEST_DATA` even if the current test run will not require any data to be downloaded?

Reply:
> Those paths could exist from a past test run, but that doesn't mean that test datasets are available for the current run. 

good point

> Is the concern that we would emit a message about GDAL_DOWNLOAD_TEST_DATA even if the current test run will not require any data to be downloaded?

changing the working to "tests relying on downloaded data may be skipped" looks good to me

Reply:
Updated in [0bfe1ad](https://github.com/OSGeo/gdal/pull/7262/commits/0bfe1ad234274ae54cc5967589770bebdb8479de)